### ISSUE TITLE
Dynamic Cells With TextViews

### ISSUE THREAD

**Issue by Alarson93** [hackiftekhar/IQKeyboardManager#497]
I have a table view. It has cells that contain textviews (constrained to the edges of the cell; scrolling disabled) and I am using automatic dimensions.

I have the following textview delegate method:
`- (void) textViewDidChange:(UITextView *) textView
{
    self.answerObject.comments = textView.text;

    __weak typeof(self) weakSelf = self;
    [UIView performWithoutAnimation:^{
         [weakSelf.tableView beginUpdates];
         [weakSelf.tableView endUpdates];
     }];
}`

Calling the begin/end updates will update the height of the cell. In iOS 8.4, IQKeyboardManager would push the cell up so the bottom line would always be at the top of the keyboard. Everything worked perfect.

Today I noticed that in 9.3, when I reach the end of the line and the next character wraps (and the cell updates height accordingly), the cell will drop instead of go up. When I type the next character (so second character of a new line) it will jump up to the expected offset.

In summary, on dynamically sizing cells with a textview, the first character of a new line will experience difficulty. 

Additionally, I notice that using return in this cell will not adjust the offset automatically. I notice that you do have documentation for a return key handler (presumably to update offset when return is pressed?), but it appears that this is targeted at view controllers. Is there anyway to update for return keys within something like a cell? (I can create a separate issue for this if you would like)

**Comment by hackiftekhar** [comment]
I undeesrand what you are trying to explain. Could you please share a demo project so that I can debug it about the first character of new line issue?

Btw return key handler is different from what you undersrand.

In general it is hard to fix issues with UITableView/UICollectionView because they usee cell reusability concept.

**Comment by Alarson93** [comment]
I will provide you with a demo later today or sometime tomorrow. 

For what it's worth, I did recently play with the estimated row height and changing that value does impact behavior.

**Comment by Alarson93** [comment]
I've attached a demo that illustrates the issue.
Make sure that you are using a 9.3 simulator (I couldn't reproduce the issue in 8.4).

[Dynamic Textview Demo.zip](https://github.com/hackiftekhar/IQKeyboardManager/files/237680/Dynamic.Textview.Demo.zip)

**Comment by Alarson93** [comment]
@hackiftekhar I saw you post in another issue that the adjustFrame method updates the position of the textView. I made adjustFrame public and called it right after endUpdates (which seems to have temporarily solved my problem until a proper solution is found).

`    __weak typeof(self) weakSelf = self;
    [UIView performWithoutAnimation:^{
         [weakSelf.tableView beginUpdates];
         [weakSelf.tableView endUpdates];
         [[IQKeyboardManager sharedManager] adjustFrame];
     }];`

**Comment by hackiftekhar** [comment]
Now I've added `reloadLayoutIfNeeded` method for you, so you can call it to refresh keyboard/textView layout.

**Comment by doingself** [comment]
Wǒ shì zhème chǔlǐ de
7/5000
I am dealing with this.```
UIView.setAnimationsEnabled(false)
tableView.beginUpdates()
tableView.endUpdates()
tableView.scrollToRow(at: indexPath, at: UITableViewScrollPosition.bottom, animated: false)
UIView.setAnimationsEnabled(true)
```
